spec/README.md

```diff
@@ -0,0 +1,100 @@
+- QGCS module / code that reads the JSON and interacts with the user-interface
+
+### HTTP / network module
+The core behavior of the internals of the adapter will be using HTTP REST based services. A high performance HTTP module that makes requests and processes responses is needed to handle the interaction with the USSP (per the config file)
+
+__Deliverables__:
+- A REST based component that interacts with the USSP APIs
+
+### Notifications
+Interaction with USSP and USSP network is a synchronous process and there needs to be a notification mechanism that tells the operator "prominently" what is going on in the UTM system. There is no standard for providing notifications to the operators but since some of the notifications will be safety critical. We are cognizant that there might be a standard that will be developed and the notifications mechanism might need to be adapted to the standard. 
+
+## Security Considerations 
+The UTM standards specify a baseline security mechnism using "internet-grade" security protocols e.g. OAUTH Client Credentials to access UTM system, additional security layers such as message-signing etc. are considered "advanced" for UTM, although they are rapidly changing. These 
+
+## Services
+For the scope of this project, we will focus on the following five services. In the future additional services might be added to the adapter. To get sense of all the services a UTM system might provide, please review [this infographic](https://www.gpsworld.com/wp-content/uploads/2020/03/U-Space-SESAR-JC-UAV-EC-report.jpg).
+
+## A. Interface for Flight Authorization
+__Goal__: Allow the GCS to send a flight plan to the USSP and get an associated flight authorization of submission of that plan.
+
+__Pre-requisites__: 
+- A flight plan is built in the QGCS user-interface and it is valid 
+- A network connection exists between QGCS and USSP
+- Metadata e.g. Operator ID etc. exists that will be passed to the USSP system
+
+__Reference__: 
+- Example submission of a plan [as GeoJSON](https://github.com/openskies-sh/flight-blender/blob/master/api/flight-blender-1.0.0-resolved.yaml#L1402)
+- Example states of an operation [Flight Blender](https://github.com/openskies-sh/flight-blender/blob/master/api/flight-blender-1.0.0-resolved.yaml#L1885)
+
+__Deliverables__:
+- User interface in the QGCS panels to submit flight plan and get the responses
+
+## B. Interface for Network RID / Telemetry
+
+__Goal__: Allow the GCS to send live telemetry (for a registered operation) to the USSP via this QGCS user-interface so that the USSP can process the data and share it with the UTM network. 
+__Pre-requisites__: 
+
+- A plan is built and shared to the USSP and authorized by the USSP (see Flight Authorization service below). 
+- Telemetry data from the drone should be submitted via TCP / IP with REST interfaces to the USSP that can be configured
```

Comment:
Is any direct drone-USSP communication envisioned here?

Reply:
I think that for this out of scope for the moment, I will add this note here.